WordPress/gutenberg
# Add Layout controls to children of Flex layout blocks

## Why?


Closes #44467

* Adds a "width" or a "height" control in the Dimensions panel of children of Row/Stack blocks (width for Row, height for Stack);
* The section contains a toggle with three options: Fit, Fill and Fixed.
* If Fixed is selected, an input is also shown, where a fixed size for the block can be set.


Follow up tasks (not within the scop of this PR):

* With vertical orientation, "Fill" won't work unless the parent container has a fixed height. Given #45300, we are now able to do this! But it might be good to add a warning or additional info on how it works.
* Should we disallow "min-height" in blocks that get the "height" control due to being Stack children?
* Revisit `safecss_filter_attr` to allow `box-sizing` rules to be output.
* look into removing styles once a block is moved out of its flex parent (or parent is transformed into a non-flex group).


## How?


* Adds an optional `allowSizingOnChildren` layout attribute to enable width/height controls on children of flex blocks.
* This attribute is only added to Row and Stack group variations for now.
* Adds an `__unstableParentLayout` prop to BlockListBlock, which is used to decide whether those child controls should be output or not, based on the existence of `allowSizingOnChildren` and a flex type layout.

## Testing Instructions




1. Add a Row block to a post and add some child blocks in it;
2. Check that width controls renders in Dimensions in the block sidebar for the child blocks;
3. Check that setting "fill" and "fixed" sizes works for child blocks.
4. Repeat test with a Stack block, which should show the "height" control on its children;
5. In order for "fill" to work for Stack children, define a min-height on the Stack block (also under Dimensions).

## Screenshots or screencast 

<img width="998" alt="Screenshot 2022-11-24 at 10 03 10 am" src="https://user-images.githubusercontent.com/8096000/203660648-983d3377-3cdf-4233-ae54-1ce1651d3a73.png">
